### PR TITLE
Add support for loading multiple corpora

### PR DESCRIPTION
**Pull request by maciejwie**
Scan code/api/corpora and load all directories as corpora, with directory name as corpus name.
Extend API to allow passing `corpus` and `max_results` parameters, with default values matching existing `"lectures"` and `5`.

Note that this doesn't add any new corpora, just adds support for adding them.

### CODE REVIEW

**Comment by maciejwie** [comment]
One thing I don't love here is that I "had" to hardcode the path to the full-text video transcripts. The segmented transcripts don't have anywhere where the full file exists though it's also something that we can add. As it is it's kind of a hack, but it works for cs410 content.

File: code/api/documents/views.py [hunk 1]
```diff
@@ -6,8 +7,34 @@ def index(request):
...
-    search_engine = engine.Engine()
+    # TODO: Add bounds checking on inputs
 
-    docs = search_engine.query_corpus(query_params.get("search", ""), 5)
+    # "search" is the only mandatory parameter
+    # Set defaults for any unset parameters
+    if query_params.get("corpus", "") == "" or query_params.get("corpus", "") == None:
+        corpus = "lectures"
+    else:
+        corpus = query_params.get("corpus", "")
```
Comment (rsuggs-uiuc-mcs) [inline code/api/documents/views.py:17]: Would it be possible to shorten this to something like this?

```python
corpus = query_params.get('corpus', 'lectures') or 'lectures'
```

I ran a few tests in the python console:

```python
>>> hash = {'test_empty_string': '', 'test_none': None}
>>> hash.get('test_empty_string')
''
>>> hash.get('test_empty_string', 'default')
''
>>> hash.get('test_empty_string', 'default') or 'default'
'default'
>>> hash.get('test_none')
>>> hash.get('test_none', 'default')
>>> hash.get('test_none', 'default') or 'default'
'default'
>>> hash.get('test_not_exist')
>>> hash.get('test_not_exist', 'default')
'default'
>>> hash.get('test_not_exist', 'default') or 'default'
'default'
```

File: code/api/lecture_search_cli.py [hunk 2]
```diff
@@ -2,11 +2,20 @@
 from pprint import pprint
 from lecture_search import engine
 
-max_results = 5  # make extensible?
-query_txt = str(input("Enter CS410 search query: "))
+query_txt = str(input("Enter search query: "))
+
+corpus = str(input("Enter corpus [lectures]: "))
```
Comment (rsuggs-uiuc-mcs) [inline code/api/lecture_search_cli.py:7]: Would this work here?

```python
corpus = str(input("Enter corpus [lectures]: ")) or 'lectures'
```

File: code/api/documents/views.py (hunk 1 above)
Comment (maciejwie) [reply code/api/documents/views.py:17]: Absolutely, that's a much better approach.

File: code/api/lecture_search_cli.py (hunk 2 above)
Comment (maciejwie) [reply code/api/lecture_search_cli.py:7]: As above, you point out a good approach and I'll make this change to both parameters.
